### PR TITLE
Adding evaluate method for MLM models

### PR DESCRIPTION
**Pull request by arunppsg**
## Description

In the current setup, evaluate does not work for masked langauge models since the input to mlm models are masked tokens which are generated on the fly and hence the labels are not known beforehand. An evaluate method in mlm will be useful for validating different mlm models. I have addressed this issue in the pull request by overriding the default `evaluate` method for masked langauge models and added tests for it.

Other changes: setting `max_length` as 100 (a hyperparameter) during tokenization of smiles strings to use in 
huggingface models and enabling padding and truncation. If the max_length variable is not set, input tokens can 
be of various length.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this 
project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by rbharath — COMMENTED** [review]
Have a request for more comments below

File: deepchem/models/torch_models/hf_models.py [hunk 1]
```diff
@@ -472,3 +477,51 @@ def _predict(self, generator: Iterable[Tuple[Any, Any, Any]],
...
         else:
             return np.array(final_results)
+
+    def evaluate(self,
+                 dataset: Dataset,
+                 metrics: List[Metric],
+                 transformers: List[Transformer] = [],
+                 per_task_metrics: bool = False,
+                 use_sample_weights: bool = False,
+                 n_classes: int = 2):
```
[the remark this answers is not in the document]
Comment (arunppsg) [reply deepchem/models/torch_models/hf_models.py:487]: added docstring and usage example

File: deepchem/models/torch_models/hf_models.py [hunk 2]
```diff
@@ -472,3 +477,86 @@ def _predict(self, generator: Iterable[Tuple[Any, Any, Any]],
...
+        >>> tokenizer = RobertaTokenizerFast.from_pretrained('seyonec/PubChem10M_SMILES_BPE_60k')
+        >>> config = RobertaConfig(vocab_size=tokenizer.vocab_size)
+        >>> model = RobertaForMaskedLM(config)
+        >>> hf_model = HuggingFaceModel(model=model, tokenizer=tokenizer, task='mlm')
+
+        >>> metrics = [dc.metrics.Metric(dc.metrics.accuracy_score)]
+        >>> scores = hf_model.evaluate(dataset, metrics=metrics)
+        """
+        self.model.train(False)
+        if self.task == 'mlm':
```
Comment (rbharath) [inline deepchem/models/torch_models/hf_models.py:524]: Thinking about this more, it feels like the more general improvement we need to make is to introduce the task API and then support metrics for more general tasks. This current code is specialized for MLM.

**Review by rbharath — COMMENTED** [review]
Thinking about it, I think this function should be a utility in the chemberta repo rather than an addition to the HFModel class since it's narrow in implementation.

**Comment by arunppsg** [comment]
I think we need a way to evaluate MLM models in deepchem at least for the time period until we introduce the task api.

**Comment by arunppsg** [comment]
As discussed offline, evaluate method will be moved chemberta3 repo.
